src/contracts/admin.requirements/admin.requirement.ts

```diff
@@ -0,0 +1,13 @@
+import { Forbidden, IRequirement } from '@panenco/papi';
+import { Request } from 'express';
+import { MyToken } from './token.type';
+
+export const adminRequirement: IRequirement = (req: Request) => {
+  const { token } = req as any;
+
+  if ((token as MyToken).role === 'admin') {
+    return;
+  }
+
+  throw new Forbidden('NoAccessToUser', 'Not enough rights');
+};
```

Comment:
You can use the declaration file to add a `token` to the `Request` interface

https://www.typescriptlang.org/docs/handbook/modules.html#ambient-modules

You can create a file like: 
`express.d.ts` and include it in the tsconfig
```ts
import {MyToken} from ...

declare module 'express' {
  interface Request {
    token: MyToken;
  }
}

```